documents/write_acls_test.go

```diff
@@ -66,24 +66,30 @@ func TestWriteACLs_getChangedFields_with_core_document(t *testing.T) {
 	doc, err := newCoreDocument()
 	assert.NoError(t, err)
 	doc.Document.DocumentRoot = utils.RandomSlice(32)
-	ndoc, err := doc.PrepareNewVersion([]string{testingidentity.GenerateRandomDID().String()}, true)
+	ndoc, err := doc.PrepareNewVersion([]string{testingidentity.GenerateRandomDID().String()}, true, nil)
 	assert.NoError(t, err)
 
 	// preparing new version would have changed the following properties
 	// current_version
 	// previous_version
 	// next_version
 	// previous_root
-	// roles
+	// roles (x2)
 	// current pre image
 	// next pre image
 	// read_rules.roles
 	// read_rules.action
+	// transition_rules.RuleKey
+	// (transition_rules.Roles
+	// transition_rules.MatchType
+	// transition_rules.Action
+	// transition_rules.Field) x 2
 	oldTree := getTree(t, &doc.Document)
 	newTree := getTree(t, &ndoc.Document)
 	cf := getChangedFields(oldTree, newTree, proofs.DefaultSaltsLengthSuffix)
-	assert.Len(t, cf, 9)
+	assert.Len(t, cf, 20)
```

Comment:
how come we increased the leaves by 11 here?

Reply:
there are 10 new fields from the two new transition rules, 2 new files from the new read rule, and two new roles. this is in addition to the 6 changed fields for the non-rule/role related changes, adding up to 20.